src/schema/filter/InputPrincipalFilterSchema.ts

```diff
@@ -0,0 +1,245 @@
+/**
+ * Input Schema of Principal
+ * @author Simpli CLI generator
+ */
+import {DefaultSchema} from '@/schema/DefaultSchema'
+import {FieldSet, FieldComponent} from '@simpli/meta-schema'
+import {InputCheckbox, InputSelect, InputText} from '@simpli/vue-input'
+import {Principal} from '@/model/resource/Principal'
+import {GrupoDoPrincipalCollection} from '@/model/collection/GrupoDoPrincipalCollection'
+import {IPrincipalCollectionResourcesHolder} from '@/model/collection/PrincipalCollection'
+
+/* TODO: review generated schema */
+export class InputPrincipalFilterSchema extends DefaultSchema
+  implements IPrincipalCollectionResourcesHolder {
+  collectionGrupoDoPrincipal = new GrupoDoPrincipalCollection().noPagination()
+
+  readonly name = 'InputPrincipalFilter'
+
+  readonly fieldSet: FieldSet<Principal> = {
+    grupoDoPrincipal1: (schema): FieldComponent => ({
+      is: InputSelect,
+      bind: {
+        items: this.collectionGrupoDoPrincipal.items,
+        multiple: true,
+        closeOnSelect: false,
+        clearOnSelect: false,
+        preserveSearch: true,
+        label: this.translateFrom(schema.fieldName),
+        trackBy: 'idGrupoPrincipalPk',
+        preselectFirst: true,
+      },
+    }),
+    grupoDoPrincipal2: (schema): FieldComponent => ({
+      is: InputSelect,
+      bind: {
+        items: this.collectionGrupoDoPrincipal.items,
+        multiple: true,
+        closeOnSelect: false,
+        clearOnSelect: false,
+        preserveSearch: true,
+        label: this.translateFrom(schema.fieldName),
+        trackBy: 'idGrupoPrincipalPk',
+        preselectFirst: true,
+      },
+    }),
+    minInteiroObrigatorio: (schema): FieldComponent => ({
+      is: InputText,
+      bind: {
+        type: 'number',
+        maxlength: 11,
+        step: 1,
+        label: this.translateFrom(schema.fieldName),
+      },
+    }),
+    maxInteiroObrigatorio: (schema): FieldComponent => ({
+      is: InputText,
+      bind: {
+        type: 'number',
+        maxlength: 11,
+        step: 1,
+        label: this.translateFrom(schema.fieldName),
+      },
+    }),
+    minDecimalObrigatorio: (schema): FieldComponent => ({
+      is: InputText,
+      bind: {
+        type: 'number',
+        maxlength: 255,
+        step: 'any',
+        label: this.translateFrom(schema.fieldName),
+      },
+    }),
+    maxDecimalObrigatorio: (schema): FieldComponent => ({
+      is: InputText,
+      bind: {
+        type: 'number',
+        maxlength: 255,
+        step: 'any',
+        label: this.translateFrom(schema.fieldName),
+      },
+    }),
+    booleanoObrigatorio: (schema): FieldComponent => ({
+      is: InputCheckbox,
```

Comment:
Acabei de perceber que esse componente não vai servir pra filtrar, pois depois q vc ativa ele não dá pra fazer ele voltar pro null nunca mais, é pra sempre true ou false, precisamos criar um outro componente pra isso

Reply:
We can review this component in the future and improve it